python/dgl/dataloading/shadow.py

```diff
@@ -0,0 +1,82 @@
+"""ShaDow-GNN subgraph samplers."""
+from ..utils import prepare_tensor_or_dict
+from ..base import NID
+from ..sampling import sample_neighbors
+from .neighbor import NeighborSamplingMixin
+from .dataloader import exclude_edges, Sampler
+
+class ShaDowKHopSampler(NeighborSamplingMixin, Sampler):
+    """K-hop subgraph sampler used by
+    `ShaDow-GNN <https://openreview.net/forum?id=GIeGTl8EYx>`__.
+
+    It performs node-wise neighbor sampling but instead of returning a list of
+    MFGs, it returns a single subgraph induced by all the sampled nodes.
+
+    This is used in conjunction with :class:`dgl.dataloading.pytorch.NodeDataLoader`
+    and :class:`dgl.dataloading.pytorch.EdgeDataLoader`.
+
+    Parameters
+    ----------
+    fanouts : list[int] or list[dict[etype, int] or None]
+        List of neighbors to sample per edge type for each GNN layer, starting from the
+        first layer.
+
+        If the graph is homogeneous, only an integer is needed for each layer.
+
+        If None is provided for one layer, all neighbors will be included regardless of
+        edge types.
+
+        If -1 is provided for one edge type on one layer, then all inbound edges
+        of that edge type will be included.
```

Comment:
I feel the description for `fanouts` needs to be improved. 
1. Why not stick to `-1` or `None` for both cases?
2. Prepare two paragraphs, one for homogeneous graphs and one for heterogeneous graphs.
3. If `fanouts` is a `list[int]`, is `fanouts[0]` for the first GNN layer or the last GNN layer? Even if users have read the user guide, they may not clearly remember that. So it's better to spell it out.
4. Provide concrete examples below for the different cases mentioned here.

Reply:
I simplified the interface so that there's only -1.  None is no longer accepted.  A minor breaking change though as I don't think people uses None explicitly instead of using `MultiLayerFullNeighborSampler`.